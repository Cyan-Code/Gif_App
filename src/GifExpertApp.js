import React, { useState } from 'react'
import { AddCategory } from './components/AddCategory'
import { GiffGrid } from './components/GiffGrid'

export const GifExpertApp = ( {defaultCategory = []} ) => {
  const [categories, setCategories] = useState( defaultCategory )  
    return (
      <>
        <h2>GifExpertApp</h2>
          <AddCategory  setCategories={ setCategories }/>
        <hr/>
        <ol>
          {
            categories.map( category => (
              <GiffGrid
                key = { category }
                category = { category }
              />
            ))
          }
        </ol>
      </>
    )
}

export default GifExpertApp
