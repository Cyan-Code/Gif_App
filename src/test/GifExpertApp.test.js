import React from 'react';
import {shallow} from 'enzyme';
import GifExpertApp from '../GifExpertApp'

describe('Pruebas sobre <GifExpertApp />', () => {
  test('Debe coincidir con el Snapshot', () => {
    
    const wrapper = shallow (<GifExpertApp/>)
    expect ( wrapper ).toMatchSnapshot()

  })

  test('Debe de mostrar una lsta de categorias', () => {
    const categories = ['One Punch','Dragon Ball']
    const wrapper = shallow (<GifExpertApp defaultCategory={categories}/>)

    expect(wrapper).toMatchSnapshot()
    expect(wrapper.find('GiffGrid').length).toBe(categories.length)
    
  })
  

})

