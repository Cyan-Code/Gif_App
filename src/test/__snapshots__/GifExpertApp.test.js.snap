// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas sobre <GifExpertApp /> Debe coincidir con el Snapshot 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;

exports[`Pruebas sobre <GifExpertApp /> Debe de mostrar una lsta de categorias 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GiffGrid
      category="One Punch"
      key="One Punch"
    />
    <GiffGrid
      category="Dragon Ball"
      key="Dragon Ball"
    />
  </ol>
</Fragment>
`;
